packages/orbit-components/src/Popover/components/ContentWrapper.tsx

```diff
@@ -64,39 +77,39 @@ const PopoverContentWrapper = ({
   const scrollingElementRef = React.useRef<HTMLDivElement | null>(null);
   useLockScrolling(scrollingElementRef, lockScrolling && !isLargeMobile);
 
-  const popoverRef = React.useRef<HTMLDivElement | null>(null);
   const windowHeight = typeof window !== "undefined" ? window.innerHeight : 0;
 
-  const { styles, update } = usePopper(referenceElement, popoverRef.current, {
-    placement,
+  const isAutoPlacement = !isFixedPlacement(placement);
+
+  const { refs, strategy, floatingStyles, context } = useFloating({
+    placement: isAutoPlacement ? undefined : placement,
     strategy: fixed ? "fixed" : "absolute",
-    modifiers: [
-      {
-        name: "offset",
-        enabled: !!offset,
-        options: {
-          offset: [offset.left, overlapped ? -Number(referenceElement?.offsetHeight) : offset.top],
-        },
-      },
-      {
-        name: "flip",
-        enabled: !noFlip,
-      },
-      { name: "preventOverflow", enabled: !allowOverflow },
+    whileElementsMounted: autoUpdate,
+    elements: {
+      reference: referenceElement,
+    },
+    middleware: [
+      offset({
+        mainAxis: overlapped ? -Number(referenceElement?.offsetHeight) : offsetProp.top,
+        crossAxis: offsetProp.left,
+      }),
+      isAutoPlacement &&
+        autoPlacement({
+          alignment: getAutoAlignment(placement),
+          autoAlignment: !noFlip,
+        }),
+      !noFlip && !isAutoPlacement && flip(),
+      !allowOverflow && shift(),
     ],
   });
 
-  const { popper } = styles;
-
   React.useEffect(() => {
     const timer = setTimeout(() => {
-      if (popoverRef.current) {
-        popoverRef.current.focus();
+      if (refs.floating.current) {
```

Comment:
I don't have access to `elements` here 🤔 I need the ref in `refs.floating` given to me by the hook

Reply:
But `elements` can be returned from the `useFloating` the same as `refs`, no?

I am asking because in the docs it's written:

> For usage in Effects, prefer using `elements.floating`, since it’s not guaranteed the floating element will be available on the first pass.

So that's why I was curious why you chose the `refs.floating`.

Reply:
Alright, I completely missed that. I will follow the docs and change it. Thank you, well spotted!

Reply:
You forgot to replace it also in the dependency array 😅

I didn't mention it in other files because I didn't want to open the same discussion in multiple places and I assumed if you would change it here you would change it there as well, my bad 🙈 `refs.floating.current` is also used inside useEffect in `TooltipContent` and `ErrorFormTooltip`. Not sure if you want to change it there as well but I guess it makes sense.

Reply:
It does, my bad. Changed